backend/src/router.py

```diff
@@ -1,42 +1,27 @@
 import json
 import logging
+from src.utils import to_json
 from src.utils import call_model
 from src.prompts import PromptEngine
+from src.agents import Agent, agents, agents_details
 
 prompt_engine = PromptEngine()
 
 
-def create_agent_object(agent):
-    agent_as_object = {
-        "name": agent.name,
-        "description": agent.description
-    }
-    return agent_as_object
-
-
-def convert_step_to_json(next_step):
-    try:
-        return json.loads(next_step)
-    except Exception:
-        raise Exception(f"Failed to interpret LLM next step format from step string\"{next_step}\"")
-
+def build_best_next_step_prompt(task, scratchpad):
+    return prompt_engine.load_prompt(
+        "best-next-step",
+        task=json.dumps(task, indent=4),
+        list_of_agents=json.dumps(agents_details, indent=4),
+        history=json.dumps(scratchpad, indent=4),
+    )
 
-# TODO: Create pick_agent test with mocked calls
-def pick_agent(current_task_object, next_task_object, agents_list, history):
-    logging.debug("Picking agent")
 
-    available_agents_object = [create_agent_object(agent) for agent in agents_list]
+response_format_prompt = prompt_engine.load_prompt("agent-selection-format")
 
-    # Generate prompts with tasks
-    best_next_step_prompt = prompt_engine.load_prompt(
-        "best-next-step",
-        current_task=json.dumps(current_task_object, indent=4),
-        next_task=json.dumps(next_task_object, indent=4),
-        list_of_agents=json.dumps(available_agents_object, indent=4),
-        history=history
-    )
 
-    response_format_prompt = prompt_engine.load_prompt("agent-selection-format")
+def get_plan(task, scratchpad):
```

Comment:
Nit-picky, but as we're creating an original plan and not retrieving one, could this be called `generate_plan` or `create_plan`?

Reply:
Maybe `build` like we use above in `build_best_next_step_prompt`?

Reply:
yeah good point, like the `build` suggestion for consistency